content/languages/cpp/authoring/index.md

```diff
@@ -0,0 +1,378 @@
+- C++ is statically typed, so any task that depends on dynamic typing can be difficult to translate into C++, and attempts to force a C++ kata to reflect a dynamically typed interface can lead to ideas that enforce a really bad design.
+- There are only a few additional libraries available to the C++ runner, so kata that take advantage of specialized packages installed for other languages become much more difficult to implement in C++.
+- Many features of C++ do not have direct equivalents in other popular languages: mixture of allowed paradigms, template meta-programming, unmanaged memory backed by RAII, native access to the platform and runtime, and many others. C++ kata which rely on them can be difficult to translate to other languages.
+
+
+## Kata snippets and template
+
+Due to the complicated compilation model of C++ code, paired with the fact that Codewars strictly enforces the names, amount, and meaning of source files, the code runner uses a `[template]()` (TODO: document C++ runner template) to concatenate all kata snippets into a single translation unit. This fact has various consequences on the code, some positive, and some negative: namespace pollution, symbols introduced by one snippet being visible in other snippets, etc. However, in the majority of cases, it does not affect kata in any way, and whenever possible, this behavior should be treated as an implementation detail of the C++ code runner. Kata snippets should be treated as separate source files, if possible.
+
+One consequence of using a template is that the signature of a solution function can be modified by the user in a way which can affect tests. To avoid the possibility of users tampering with the prototype of a solution function, it's recommended to re-declare it in the tests snippet.
+
+Another issue caused by the use of the template is excessive namespace pollution. See paagraph on [namespaces](#namespaces) for more details.
+
+
+## Coding
+
+### Code style
+
+C++ programmers have many sets of naming conventions or code style guides. Some of them can be found for example [here](https://isocpp.org/wiki/faq/coding-standards), or [here](https://google.github.io/styleguide/cppguide.html). Codewars does not strictly enforce any of them, just use whatever set of guidelines you like, but when you do, use it consistently. 
+
+There are a few C++ coding guidelines which are violated by kata authors and translators particularly often:
+- Arguments which are not cheap to copy should be passed by const reference, and not by value.
+- Appropriate containers should be used, _especially_ when used as input arguments or return values:
+  - C-style raw arrays and pointers to arrays need to be avoided
+  - `std::array`, `std::pair`, or `std::tuple` should be used for data of known size. Avoid representing 2D points or pairs with `std::vector`.
+  - `std::vector` should be used for arrays of varying size
```

Comment:
I don't like calling non-arrays arrays, especially talking about languages that have arrays (although I see it in Python all the time anyway).
And I think the old normal C++ way would be a pair of iterators in many cases. Now that I think about it, it may actually be a good idea to update to C++20 now and use ranges and `span`s for input.

Reply:
```suggestion
  - `std::vector` is not the only available option, and can be replaced by another container (for example, `std::set`), a set of iterators, a range, or any other construct suitable for the task.
```